Source/FakeItEasy.Specs/UnconfiguredFakeSpecs.cs

```diff
@@ -6,23 +6,28 @@
 
     public static class UnconfiguredFakeSpecs
     {
+        public interface IFoo
+        {
+            bool IsADummy { get; }
+        }
+
         [Scenario]
         public static void VirtualMethod(
             MakesVirtualCallInConstructor fake)
         {
-            "when faking a class with a virtual method"
+            "When faking a class with a virtual method"
                 .x(() => fake = A.Fake<MakesVirtualCallInConstructor>());
 
-            "it should return a default value when the method is called during the constructor"
+            "It should return a default value when the method is called during the constructor"

```

Comment:
I agree that this should be split into two scenarios.

In the case of the first scenario, I think we have three choices:

1. Construct the type via reflection (or Roslyn!) in a "Given" and "And"s (and create the fake similarly)
2. Declare the type and have an empty "Given" and "And"s
3. Declare the type and skip the "Given" and "And"s

In C#, it is way more convenient to declare the types up front rather than construct them reflectively,  and the whole scenario will look very different from the usage of the API we want to test, so we probably want to skip option 1 (this may not be the case in other languages). Option 2 is best in terms of the narrative but some people may baulk at having empty steps. Option 3 omits the empty step(s), but can result in a rather wordy When.

E.g.

```Gherkin
# Options 1 and 2
Scenario: A constructor calls a virtual method
  Given a type with a default constructor
    And the constructor calls a virtual method
    And the method returns a non-default value
   When I create a fake of the type
   Then the method returns a default value
    And the call is recorded

# Option 3
Scenario: A constructor calls a virtual method
   When I create a fake of a type with a default constructor which calls a virtual method which returns a non-default value
   Then the method returns a default value
    And the call is recorded
````

For the second scenario, I'd probably go with something like
```Gherkin
Scenario: A virtual method
  Given a fake with a virtual method
   When I call the method
   Then a default value is returned
    And the call is recorded
```

Forming these narratives can be quite an art form. I find it one of the most challenging, nevertheless interesting and valuable exercises in development. It really forces you drive out what it is you are testing and often yields invaluable feedback which can have a huge influence on the user story/feature itself.